src/lib/state.ts

```diff
@@ -72,21 +75,58 @@ export async function fetchWtnAllocation(
 	}
 }
 
+export class CanisterActor<canisterInterface> {
+	public authenticatedActor?: canisterInterface;
+	public actor: canisterInterface;
```

Comment:
I would name actor `unauthenticatedActor` to make sure that we don't confuse the two. I had an implementation using the same name as you and got a lot of confusion from that.

Reply:
Yes I thought about it

Reply:
maybe `anonymousActor` is the clearest?

Reply:
changed it